amazon/src/main/java/amazon/robot/AwaitingJobState.java

```diff
@@ -0,0 +1,77 @@
+package amazon.robot;
+
+import amazon.*;
+import simulator.*;
+import simulator.utils.*;
+
+class AwaitingJobState implements RobotState {
+  private final Robot context;
+
+  public AwaitingJobState(Robot context) {
+    this.context = context;
+  }
+
+  @Override
+  public void tick(Warehouse warehouse, int currentTickCount) throws Exception {
+    // If at Charging Pod, charge, otherwise move towards it.
+    if (this.context.getLocation().equals(this.context.chargingPod.getLocation()))
+      this.context.charge();
+    else
+      this.context.move(warehouse.getFloor(), this.context.chargingPod.getLocation());
+
+    // Check if there's an acceptable Job to pick up.
+    this.pickupJob(warehouse.getJobManager(), warehouse.getFloor());
+  }
+
+  /**
+	 * @param jobManager
+	 * @param floor
+	 * @throws Exception
+	 */
+	private void pickupJob(ItemManager<Job> jobManager, Floor floor) throws Exception {
+		if (!jobManager.areItemsToPickup())
+			return;
+
+		// Check out next Job and validate estimated cost.
+		Job tempJob = jobManager.viewNextPickup();
+		double estimatedCostWithLeeway = estimatePowerUnitCostForJob(tempJob, floor);
+		if (estimatedCostWithLeeway > this.context.powerUnits)
+			return;
+
+		// Pickup Job
+		this.context.currentJob = jobManager.pickup();
+    this.context.setPathFinder(null);
+    this.context.state = this.context.collectingItemState;
+
+		// this.log("Job to %s then %s picked up.", this.context.currentJob.getStorageShelf().getLocation(),
+		// 		this.context.currentJob.getPackingStation().getLocation());
+  }
+  
+  /**
+	 * Given a storage shelf and packing station this will calculate the number of power units to make
+	 * the trip back to its charging pod with a leeway of 20%.
+	 * 
+	 * @param storageShelf
+	 * @param packingStation
+	 * @param warehouse
+	 * @return
+	 * @throws LocationNotValidException
+	 */
+	private double estimatePowerUnitCostForJob(Job job, Floor floor) throws Exception {
+
+		// Setup Cost Estimator
+		IPathCostEstimator costEstimator = new BasicPathCostEstimator(this.context.getLocation());
```

Comment:
Normally you'd want to have something else tell you which estimator to use, rather than creating it here. In fact, it might be better to grab it from the Robot, so whoever sets up the Robot (usually the simulation) can specify which estimator the Robot will use.